components/server/src/api/workspace-service-api.ts

```diff
@@ -92,4 +99,80 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.pagination.total = resultTotal;
         return response;
     }
+
+    async createAndStartWorkspace(req: CreateAndStartWorkspaceRequest): Promise<CreateAndStartWorkspaceResponse> {
+        // We rely on FGA to do the permission checking
+        if (req.source?.case !== "contextUrl") {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "not implemented");
+        }
+        if (!req.organizationId || !uuidValidate(req.organizationId)) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "organizationId is required");
+        }
+        if (!req.editor) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "editor is required");
+        }
+        if (!req.source.value) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "value is required in contextUrl source");
+        }
+        const contextUrl = req.source.value;
+        const user = await this.userService.findUserById(ctxUserId(), ctxUserId());
+        const { context, project } = await this.contextService.parseContext(user, contextUrl, {
+            projectId: req.configurationId,
+            organizationId: req.organizationId,
+            forceDefaultConfig: req.forceDefaultConfig,
+        });
+
+        await this.workspaceService.mayStartWorkspace({}, user, req.organizationId, Promise.resolve([]));
```

Comment:
Hm, I think this should go into the WorkspaceService.createWorkspace instead... :thinking:

Reply:
Not sure why I did not move it when introducing WorkspaceService. I think I could not do it due to PR dependencies, and did not clean up afterwards :grimacing: